### ISSUE TITLE
`unified-signatures` erroneously claims overloads can be combined

### ISSUE THREAD

**Issue by pelotom** [palantir/tslint#2338]
### Bug Report

- __TSLint version__: 4.5.1
- __TypeScript version__: 2.2.1
- __Running TSLint via__: CLI, VSCode

#### TypeScript code being linted

```ts
function foo(...xs: number[]): void
function foo(xs: number[], y: string): void
function foo(...args: any[]) {
  // ...
}
```

#### Actual behavior
```
function foo(xs: number[], y: string): void
                           ^^^^^^^^^
These overloads can be combined into one signature with an optional parameter.
```

#### Expected behavior

There should be no warning here, because these signatures cannot be further combined. The first signature uses varargs, but the second cannot due to the extra `y` parameter.
